File: elucidate-server/src/test/java/com/digirati/elucidate/test/model/enumeration/ClientPreferenceTest.java
package com.digirati.elucidate.test.model.enumeration;

import com.digirati.elucidate.model.enumeration.ClientPreference;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertNotNull;

public class ClientPreferenceTest {

    @Test
    public void testClientPreferenceMinimalContainer() {
        ClientPreference clientPreference = ClientPreference.MINIMAL_CONTAINER;
        assertNotNull(clientPreference);
        assertThat(clientPreference, is(equalTo(ClientPreference.MINIMAL_CONTAINER)));
    }

    @Test
    public void testClientPreferenceContainedIris() {
        ClientPreference clientPreference = ClientPreference.CONTAINED_IRIS;
        assertNotNull(clientPreference);
        assertThat(clientPreference, is(equalTo(ClientPreference.CONTAINED_IRIS)));
    }

    @Test
    public void testClientPreferenceContainedDescriptions() {
        ClientPreference clientPreference = ClientPreference.CONTAINED_DESCRIPTIONS;
        assertNotNull(clientPreference);
        assertThat(clientPreference, is(equalTo(ClientPreference.CONTAINED_DESCRIPTIONS)));
    }
}

File: elucidate-server/src/test/java/com/digirati/elucidate/test/model/enumeration/ClientPreferenceTest.kt
package com.digirati.elucidate.test.model.enumeration

import com.digirati.elucidate.model.enumeration.ClientPreference
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.Assert
import org.junit.Test

class ClientPreferenceTest {
    @Test
    fun testClientPreferenceMinimalContainer() {
        val clientPreference = ClientPreference.MINIMAL_CONTAINER
        Assert.assertNotNull(clientPreference)
        MatcherAssert.assertThat(clientPreference, Matchers.`is`(Matchers.equalTo(ClientPreference.MINIMAL_CONTAINER)))
    }

    @Test
    fun testClientPreferenceContainedIris() {
        val clientPreference = ClientPreference.CONTAINED_IRIS
        Assert.assertNotNull(clientPreference)
        MatcherAssert.assertThat(clientPreference, Matchers.`is`(Matchers.equalTo(ClientPreference.CONTAINED_IRIS)))
    }

    @Test
    fun testClientPreferenceContainedDescriptions() {
        val clientPreference = ClientPreference.CONTAINED_DESCRIPTIONS
        Assert.assertNotNull(clientPreference)
        MatcherAssert.assertThat(
            clientPreference,
            Matchers.`is`(Matchers.equalTo(ClientPreference.CONTAINED_DESCRIPTIONS))
        )
    }
}